src/Controls/src/SourceGen/CodeBehindGenerator.cs

```diff
@@ -125,7 +125,14 @@ static void GenerateXamlCodeBehind(ProjectItem projItem, Compilation compilation
 			var text = projItem.AdditionalText.GetText();
 			if (text == null)
 				return;
-			if (!TryParseXaml(text, compilation, xmlnsDefinitionCache, out var rootType, out var rootClrNamespace, out var generateDefaultCtor, out var addXamlCompilationAttribute, out var hideFromIntellisense, out var XamlResourceIdOnly, out var baseType, out var namedFields, out var parseException))
+
+			// Get a unique string for this xaml project item
+			var itemName = projItem.ManifestResourceName ?? projItem.RelativePath;
+			if (itemName == null)
+				return;
+			var uid = Crc64.ComputeHashString(itemName);
```

Comment:
I'd worry if a CRC is collision-resistant enough compared to something more aggressive, but this should help,

Reply:
@jonathanpeppers  you did some evaluation of hashing algorithms when you decided on CRC64 for android uses (and so we followed suit for MAUI to replace our non-security related uses of MD5).  What's your risk assessment?

Reply:
We use it to get a Java package name for a C# namespace, and we also had the constraint of needing it to be short. In our case it would have been two C# types with the same name, and also somehow hit a hash collision on the namespace.

In this case, maybe it would be easier to find a collision, any two paths just would have to collide? But up to you.

Reply:
Added the `context.AssemblyName` to the hash... That should be enough hopefully to mitigate collisions?